### PR TITLE
feat(metrics): add volumes metrics support for raw block volume

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateek.pandey@mayadata.io>

PR adds the csi volums metrics support for block volumes , so that these matrics can me exposed via promethus running in a cluster

### CODE REVIEW

File: pkg/driver/node.go [hunk 1]
```diff
@@ -438,3 +457,16 @@ func (ns *node) GetStatistics(volumePath string) (VolumeStatistics, error) {
...
+
+func (ns *node) getBlockSizeBytes(devicePath string) (int64, error) {
+	output, err := ns.mounter.Exec.Command("blockdev", "--getsize64", devicePath).CombinedOutput()
+	if err != nil {
+		return -1, fmt.Errorf("error when getting size of block volume at path %s: output: %s, err: %v", devicePath, string(output), err)
+	}
+	strOut := strings.TrimSpace(string(output))
+	gotSizeBytes, err := strconv.ParseInt(strOut, 10, 64)
+	if err != nil {
+		return -1, fmt.Errorf("failed to parse size %s into int a size", strOut)
```
Comment (utkarshmani1997) [inline pkg/driver/node.go:469]: Should you return error also?

File: pkg/driver/node.go [hunk 2]
```diff
@@ -394,6 +395,24 @@ func (ns *node) NodeGetVolumeStats(
...
+		bcap, err := ns.getBlockSizeBytes(volumePath)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get block capacity on path %s: %v", req.VolumePath, err)
+		}
+		return &csi.NodeGetVolumeStatsResponse{
+			Usage: []*csi.VolumeUsage{
+				{
+					Unit:  csi.VolumeUsage_BYTES,
+					Total: bcap,
+				},
```
[the remark this answers is not in the document]
Comment (prateekpandey14) [reply pkg/driver/node.go:412]: Yes with raw black device we don't have `used` and `Available` details

File: pkg/driver/node.go (hunk 1 above)
Comment (prateekpandey14) [reply pkg/driver/node.go:469]: added the error in return

**Comment by sonasingh46** [comment]
@prateekpandey14 -- Can you please cherry-pick this to 2.2.x branch?

**Pull request merged**